sky/backends/backend_utils.py

```diff
@@ -994,7 +998,8 @@ def write_cluster_config(
         dict(
             resources_vars,
             **{
-                'cluster_name': cluster_name,
+                'cluster_name': cluster_name_on_cloud,
+                'user_cluster_name': cluster_name,
```

Comment:
How about naming the two keys `cluster_name_on_cloud`, `cluster_name`, for consistency?